### PR TITLE
EVEREST-357 reduce cluster name size limit to 22

### PR DESCRIPTION
**Pull request by recharte**
EVEREST-357

Prevent resource names exceeding RFC1123 size limits by limiting the cluster name to 22 characters.

### CODE REVIEW

**Comment by michal-kralik** [comment]
Is this going to help? The name in the ticket had 21 characters.

File: api/validation.go [hunk 1]
```diff
@@ -54,8 +54,16 @@ func ErrInvalidURL(fieldName string) error {
...
-		return errors.Errorf("'%s' can be at most 63 characters long", name)
+	// We are diverging from the RFC1123 spec in regards to the length of the
+	// name because the operators use the name as a sufix for other resources
+	// which also have a limit of 63 characters. We want to avoid the situation
+	// where the name of the resource is longer than the limit.
+	// The longest resource name seems to be the pxc restore job name which is
+	// restore-job-<restore-cr-name>-<cluster-name>. The restore-cr-name is the
+	// same as the cluter-name, so we limit this to 25 characters calculated
+	// from (63 - len("restore-job-"))/2.
+	if len(s) > 25 {
```
Comment (hors) [inline api/validation.go:65]: @recharte in PXC operator we have limit 22 https://github.com/percona/percona-xtradb-cluster-operator/blob/main/pkg/apis/pxc/v1/pxc_types.go#L618

**Review by hors — CHANGES_REQUESTED** [review]

**Review by gen1us2k — APPROVED** [review]
Thanks. Looks good but please resolve changes requested by Slava

**Review by oksana-grishchenko — APPROVED** [review]
Approving to unblock the PR once Slava's comment is resolved

**Comment by recharte** [comment]
> Is this going to help? The name in the ticket had 21 characters.

This other improvement gave us some more characters https://github.com/percona/everest-operator/pull/61.

File: api/validation.go (hunk 1 above)
Comment (recharte) [reply api/validation.go]: Thank you @hors

**Review by hors — APPROVED** [review]

**Pull request merged**